website/source/api/auth/gcp/index.html.md

```diff
@@ -0,0 +1,362 @@
+service account or other Google public certificates to confirm signed JWTs
+passed in during login.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `POST`   | `/auth/gcp/config/`          | `204 (empty body)`     |
+
+### Parameters
+
+- `credentials` `(string: "")` - A marshaled JSON string that is the content
+  of a GCP credentials file. If not provided, the Vault server attempts
+  to use [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
+
+
+### Sample Payload
+
+```json
+{
+  "credentials": "{ \"type\": \"service_account\", \"project_id\": \"project-123456\",...}"
+}
+```
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    https://vault.rocks/v1/auth/gcp/config
+```
+
+## Read Config
+
+Returns the previously configured config, including credentials.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `GET`    | `/auth/gcp/config`           | `200 application/json` |
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    https://vault.rocks/v1/auth/gcp/config
+```
+
+### Sample Response
+
+```json
+{
+  "auth":null,
+  "warnings":[
+    "Read access to this endpoint should be controlled via ACLs as it will return the configuration information as-is, including any passwords."
+  ],
+
+  "data":{
+    "client_email":"serviceaccount1@project-123456.iam.gserviceaccount.com",
+    "client_id":"...",
+    "private_key":"-----BEGIN PRIVATE KEY-----...-----END PRIVATE KEY-----\n",
+    "private_key_id":"...",
+    "project_id":"project-123456"
+  },
+  "lease_duration":0,
+  "lease_id":"",
+  "renewable":false,
+}
+
+```
+
+## Delete Config
+
+Deletes the previously configured GCP config and credentials.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `DELETE` | `/auth/gcp/config`           | `204 (empty body)`     |
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request DELETE \
+    https://vault.rocks/v1/auth/gcp/config/client
+```
+
+## Create Role
+
+Registers a role in the backend. Role types have specific entities
+that can perform login operations against this endpoint. Constraints specific
+to the role type must be set on the role. These are applied to the authenticated
+entities attempting to login.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `POST`   | `/auth/gcp/role/:role`       | `204 (empty body)`     |
+
+### Parameters
+
+- `role` `(string: <required>)` - Name of the role.
+- `type` `(string: "iam")` - The type of this role. Only the  
```

Comment:
`(string: "iam")` -> `(string: <required>)`. The values mentioned here on the fields denote default values, and we don't default the `type` field to "iam". Instead, the API expects it to be set.